src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -741,8 +741,8 @@ private class LocalSymbolWithEnclosingContext : SourceLocalSymbol
                     nodeToBind.Kind() == SyntaxKind.CasePatternSwitchLabel ||
                     nodeToBind.Kind() == SyntaxKind.ThisConstructorInitializer ||
                     nodeToBind.Kind() == SyntaxKind.BaseConstructorInitializer ||
-                    nodeToBind.Kind() == SyntaxKind.SimpleBaseType || // initializer for a record constructor
-                    nodeToBind.Kind() == SyntaxKind.ArgumentList && nodeToBind.Parent is ConstructorInitializerSyntax ||
+                    nodeToBind.Kind() == SyntaxKind.PrimaryConstructorBaseType || // initializer for a record constructor
+                    nodeToBind.Kind() == SyntaxKind.ArgumentList && (nodeToBind.Parent is ConstructorInitializerSyntax || nodeToBind.Parent is PrimaryConstructorBaseTypeSyntax) ||
```

Comment:
>PrimaryConstructorBaseTypeSyntax [](start = 143, length = 32)

Is this tested?

Reply:
> Is this tested?

Yes, this scenario is exercised by speculative operations.

---
In reply to: [443927699](https://github.com/dotnet/roslyn/pull/45351#discussion_r443927699) [](ancestors = 443927699)

Reply:
Can you please identify a test which exercises this?

---
In reply to: [443929018](https://github.com/dotnet/roslyn/pull/45351#discussion_r443929018) [](ancestors = 443929018,443927699)

Reply:
> Can you please identify a test which exercises this?

BaseArguments_19, GetSpeculativeSymbolInfo for an initializer with a an out var in an argument.

---
In reply to: [443929431](https://github.com/dotnet/roslyn/pull/45351#discussion_r443929431) [](ancestors = 443929431,443929018,443927699)

Reply:
This line as well ``` Assert.Equal("Base..ctor(System.Int32 X)", speculativeModel!.GetSymbolInfo((SyntaxNode)speculativePrimaryInitializer).Symbol.ToTestDisplayString());```

---
In reply to: [443930921](https://github.com/dotnet/roslyn/pull/45351#discussion_r443930921) [](ancestors = 443930921,443929431,443929018,443927699)